pkg/url/url.go

```diff
@@ -396,6 +397,27 @@ func GetValidPortNumber(componentName string, portNumber int, portList []string)
 	return portNumber, fmt.Errorf("given port %d is not exposed on given component, available ports are: %s", portNumber, strings.Trim(strings.Replace(fmt.Sprint(componentPorts), " ", ",", -1), "[]"))
 }
 
+// GetValidExposedPortNumber checks if the given exposed port number is a valid port or not
+// if exposed port is not provided, a random 5 digit valid port will be generated and returned
```

Comment:
HttpGetFreePort() can assign a port which is not of 5 digits too

Reply:
updated